packages/image-block/src/components/Image.tsx

```diff
@@ -16,65 +18,87 @@ type ImageProps = {
     globalAssetViewerEnabled: boolean;
 };
 
-export const ImageComponent = ({
+type ImageWrapperProps = {
+    image: Asset;
+    appBridge: AppBridgeBlock;
+    children: ReactNode;
+    link: Link | null;
+    style: CSSProperties;
+    isEditing: boolean;
+    isAssetViewerEnabled: boolean;
+    alignment: Alignment;
+};
+
+type ImageComponentProps = {
+    image: Asset;
+    alt: string;
+    containerWidth: number;
+};
+
+const ImageWrapper = ({
     image,
-    blockSettings,
-    isEditing,
     appBridge,
+    children,
+    link,
+    style,
+    isEditing,
     isAssetViewerEnabled,
-    containerWidth,
-}: ImageProps & { isAssetViewerEnabled: boolean; containerWidth: number }) => {
+    alignment,
+}: ImageWrapperProps) => {
     const { open } = useAssetViewer(appBridge);
-    const link = blockSettings?.hasLink && blockSettings?.linkObject?.link && blockSettings?.linkObject;
-    const { isFocused, focusProps } = useFocusRing();
-
-    const Image = (
-        <ResponsiveImage
-            testId="image-block-img"
-            image={image}
-            containerWidth={containerWidth}
-            alt={blockSettings.altText ?? ''}
-            style={{ maxWidth: image.width }}
-        />
-    );
 
-    const props = {
-        ...focusProps,
+    const sharedProps = {
         className: joinClassNames([
-            'tw-rounded tw-w-full tw-flex',
-            isFocused && FOCUS_STYLE,
-            mapAlignmentClasses[blockSettings.alignment],
+            'tw-flex tw-overflow-hidden tw-w-full',
+            FOCUS_VISIBLE_STYLE,
+            mapAlignmentClasses[alignment],
         ]),
+        style,
     };
 
-    if (isEditing) {
-        return Image;
+    if (isEditing || (!link && !isAssetViewerEnabled)) {
+        return (
+            <div {...sharedProps} data-test-id="image-block-default-wrapper">
+                {children}
+            </div>
+        );
     }
 
     if (link) {
         return (
             <a
-                {...props}
+                {...sharedProps}
                 href={link.link.link}
                 target={link.openInNewTab ? '_blank' : undefined}
                 rel={link.openInNewTab ? 'noopener noreferrer' : 'noreferrer'}
+                data-test-id="image-block-link-wrapper"
             >
-                {Image}
+                {children}
             </a>
         );
     }
 
     if (isAssetViewerEnabled) {
         return (
-            <button data-test-id="image-block-asset-viewer-button" {...props} onClick={() => open(image)}>
-                {Image}
+            <button data-test-id="image-block-asset-viewer-wrapper" {...sharedProps} onClick={() => open(image)}>
+                {children}
             </button>
         );
     }
 
-    return Image;
+    return null;
```

Comment:
nit, to avoid the return null, I would prefer it like this:
```
if (link && !isEditing) return ...
if (isAssetViewerEnabled && !isEditing) return ...
return <div {...sharedProps} {...}>{children}</div>
```

Reply:
fixed